appcontainer/ViewportSizeModel.js

```diff
@@ -42,27 +34,41 @@ export class ViewportSizeModel extends HoistModel {
         makeObservable(this);
         window.addEventListener('resize', () => this.setViewportSize());
         this.setViewportSize();
+
+        this.addReaction({
+            track: () => this.size,
+            run: () => this.updateOrientation(),
+            fireImmediately: true,
+            // Further debounce (on top of debounced setViewportSize) to allow for even greater
+            // lag observed on mobile device responses to DOM orientation APIs.
+            debounce: 100
+        });
     }
 
     //---------------------
     // Implementation
     //---------------------
-    @action
+    @debounced(100)  // Debounced to account for slow inner window resize on mobile.
```

Comment:
FWIW, I've found you can use both @debounced and @action annotations on the same method, as long as you do so in that order (if you wanted to avoid the runInAction below)